src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -32,28 +32,28 @@ protected override bool IsUnusableType(ITypeSymbol otherSideType)
                     (otherSideType.Name == string.Empty || otherSideType.Name == "var");
             }
 
-            protected override IEnumerable<TypeInferenceInfo> GetTypes_DoNotCallDirectly(ExpressionSyntax expression, bool objectAsDefault)
+            protected override IEnumerable<TypeInferenceInfo> GetTypes_DoNotCallDirectly(SyntaxNode node, bool objectAsDefault)
             {
-                var types = GetTypesSimple(expression).Where(IsUsableTypeFunc);
+                var types = GetTypesSimple(node).Where(IsUsableTypeFunc);
                 if (types.Any())
                 {
                     return types;
                 }
 
-                return GetTypesComplex(expression).Where(IsUsableTypeFunc);
+                return GetTypesComplex(node).Where(IsUsableTypeFunc);
             }
 
-            private static bool DecomposeBinaryOrAssignmentExpression(ExpressionSyntax expression, out SyntaxToken operatorToken, out ExpressionSyntax left, out ExpressionSyntax right)
+            private static bool DecomposeBinaryOrAssignmentExpression(SyntaxNode node, out SyntaxToken operatorToken, out ExpressionSyntax left, out ExpressionSyntax right)
```

Comment:
why was this changed?

Reply:
Because the type inferred now works across expressions and patterns uniformly.

Reply:
I don't see how that's relevant here. This method is called DecomposeBinaryOrAssignmentExpression and still only handles expressions.

Reply:
sorry, i thought you were asking a general question about the use of Expression vs SyntaxNode.  I can make this ExpressionSyntax, though then i still have to add casts.  That's not a big deal, just something that wasn't necessary if this was typed as SyntaxNode.